datasets/emotone_ar/emotone_ar.py

```diff
@@ -0,0 +1,83 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dataset of 10065 tweets in Arabic for Emotion detection in Arabic text """
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+
+import datasets
+
+
+_CITATION = """\
+@inbook{inbook,
+author = {Al-Khatib, Amr and El-Beltagy, Samhaa},
+year = {2018},
+month = {01},
+pages = {105-114},
+title = {Emotional Tone Detection in Arabic Tweets: 18th International Conference, CICLing 2017, Budapest, Hungary, April 17–23, 2017, Revised Selected Papers, Part II},
+isbn = {978-3-319-77115-1},
+doi = {10.1007/978-3-319-77116-8_8}
+}
+"""
+
+
+_DESCRIPTION = """\
+Dataset of 10065 tweets in Arabic for Emotion detection in Arabic text"""
+
+
+_HOMEPAGE = "https://github.com/AmrMehasseb/Emotional-Tone"
+
+
+_DOWNLOAD_URL = "https://raw.githubusercontent.com/AmrMehasseb/Emotional-Tone/master/Emotional-Tone-Dataset.csv"
+
+
+class EmoToneAr(datasets.GeneratorBasedBuilder):
+    """Dataset of 10065 tweets in Arabic for Emotions detection in Arabic text"""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "ID": datasets.Value("string"),
+                    " TWEET": datasets.Value("string"),
+                    " LABEL": datasets.Value("string"),
```

Comment:
could you use the ClassLabel feature type for the LABEL field ?
For example
```
"label": datasets.ClassLabel(names=["love", ...])
```

Also can you rename the columns `id`, `tweet` and `label` ?

After this change you'll need to regenerate the dataset_infos.json file

Reply:
@lhoestq same issue here when renaming the dataset labels. See the error when I run the cli test:

```
Traceback (most recent call last):
  File "datasets-cli", line 36, in <module>
    service.run()
  File "/home/mesfas/opensource/datasets/src/datasets/commands/test.py", line 113, in run
    builder.download_and_prepare(
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 526, in download_and_prepare
    self._download_and_prepare(
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 604, in _download_and_prepare
    self._prepare_split(split_generator, **prepare_split_kwargs)
  File "/home/mesfas/opensource/datasets/src/datasets/builder.py", line 905, in _prepare_split
    example = self.info.features.encode_example(record)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 799, in encode_example
    return encode_nested_example(self, example)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 709, in encode_nested_example
    return dict(
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 710, in <genexpr>
    (k, encode_nested_example(sub_schema, sub_obj)) for k, (sub_schema, sub_obj) in utils.zip_dict(schema, obj)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 737, in encode_nested_example
    return schema.encode_example(obj)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 522, in encode_example
    example_data = self.str2int(example_data)
  File "/home/mesfas/opensource/datasets/src/datasets/features.py", line 481, in str2int
    output.append(self._str2int[str(value)])
KeyError: 'LABEL'
```

Reply:
and this is how I've rewritten the features:

```
            features=datasets.Features(
                {
                    "id": datasets.Value("string"),
                    "tweet": datasets.Value("string"),
                    "label": datasets.ClassLabel(names = ["none", "anger", "joy", "sadness", "love", "sympathy", "surprise"]),
                }
            ),
```

Reply:
did you try to also rename the field names before yielding the example at line 83 ?

Reply:
If the errors keep happening you can try to clear your cache (by default at ~/.cache/huggingface/datasets) and also delete the dataset_infos.json file and then try again

Reply:
Ok I found the issue !

This is because the first line of the CSV file contains the column names. This column has to be ignored